### DISCUSSION TITLE
Using latest value as input to generate next value

### DISCUSSION THREAD

**Discussion by KostasTsar** [bonsai-rx/bonsai#1184] [Q&A]
Hello,

Apologies for the simple question again, nothing I try works...
I am trying to figure the logic for trigerring an event at random frame intervals. Basically, sample a random number, and then when frame counter == random number: perform event. My problem arises when I try to generate subsequent random numbers. Iinitially I was thinking I should generate the next frame interval once the first frame interval is reached. However, I can't figure out how to feed the last generated value to the generation of the new value without causing a definition error (variable is defined in terms of itself). I am attaching a picture of the script below. What do I need to do to fix this, or is this logic flawed from the beginning?

Thank you in advance.

<img width="1152" alt="example" src="https://user-images.githubusercontent.com/84953213/206892975-ac6ee4f9-dd8e-4b31-bc14-47e85ed51d30.png">

**Comment by bruno-f-cruz** [comment]
In your example you need to initialize the value of some variables otherwise you will have a circular dependency. Initializing the value Time_to_Sample and use multicastsubject to update it later should work. However, for your particular needs I think you can do something simpler:

![image](https://user-images.githubusercontent.com/7049351/206899530-f4b6251d-dcb6-4d95-bf1c-a70a27d60a9c.png)

Defer subscribes to `RenderFrame `each time it starts. It sets the `Take` node property to a random number and only emits the `last` value of this sequence. Outside the `Repeat `node forces the defer to start again and repeats the loop.

**Comment by KostasTsar** [comment]
Thank you very much Bruno, this works great!
If it's not too much trouble, could you also point out an example of how to avoid the circular dependency with the multicast object? Sounds really useful.

**Reply by KostasTsar** [reply]
Nevermind, got it. Thank you, very useful tool!
